#708: Added method for vscode

leonrohne27 opened the pull request:
Fixes:#708

### implements:

- Added method that reads and passes the workspace path when calling "ide vscode", so that the user doesn't need to hand over the directory manually.

hohwille commented on cli/src/main/java/com/devonfw/tools/ide/tool/vscode/Vscode.java:
```diff
@@ -131,6 +131,7 @@ protected void configureToolArgs(ProcessContext pc, ProcessMode processMode, Pro
     pc.addArg("--user-data-dir=" + vsCodeConf);
     Path vsCodeExtensionFolder = this.context.getIdeHome().resolve("plugins/vscode");
     pc.addArg("--extensions-dir=" + vsCodeExtensionFolder);
+    pc.addArg(this.context.getWorkspacePath());
```
Did you test this in combination with plugin installations?
This argument will be added to any invocation of `runTool` and that is not only used to start the IDE but also to install plugins. In the latter case this argument is IMHO undesired and might even break VScode.
In case I am right, you could make this conditional on `processMode` or you could move this extra arg to `run()` method (overriding it).

leonrohne27 replied:
Yes I just tested it and it worked out regardless. Anyways, I've found a bug not related to my change:
When vscode is already installed, plugins won't be reinstalled when a user manually deleted the plugins from the plugins/vscode- folder. Instead it just create an empty extensions.json- file.

hohwille replied:
That is covered by #654.
Thanks for testing.

hohwille reviewed (changes requested):
@leonrohne27 thanks for your PR. You found the right way to solve this problem 👍 
https://code.visualstudio.com/docs/editor/command-line#_opening-files-and-folders

However, the place in the code where you integrated the change may be incorrect.
Could you have a look and double-check it? Thanks.

hohwille reviewed (approved):
@leonrohne27 this is fine now after your comment that resolved my doubt.
Could you please follow DoD and add this to our [changelog](https://github.com/devonfw/IDEasy/blob/main/CHANGELOG.adoc#202411001).
The first thing then would be to pull your feature branch so you have the latest CHANGELOG (I already updated your branch to main).

leonrohne27 commented:
> @leonrohne27 this is fine now after your comment that resolved my doubt. Could you please follow DoD and add this to our [changelog](https://github.com/devonfw/IDEasy/blob/main/CHANGELOG.adoc#202411001). The first thing then would be to pull your feature branch so you have the latest CHANGELOG (I already updated your branch to main).

thanks, I added the entry.
